### PR TITLE
Persist to localstorage

### LINKED ISSUE #4: Save, track changes locally
Create the ability to stash changes to local storage. 

On save, the app should serialize the changes into a format that can be uploaded and inserted into a database.

The app should clear changes on successful save, and reload to a previous state after crashing.

The app should support undo and redo features.

### PR DESCRIPTION
**Pull request by dereklieu**
Close #4 

Notes:

- For simplicity's sake, I don't store the entire history, opting instead to store a compressed version, where the entire history gets compressed into a single action. This compression utility is now part of the thing that generates the payload when we save, which makes that code simpler to write.
- Includes an extra "fast forward" action that treats the locally saved selection as essentially a `REDO` action. This assumes that this only happens at the start, which I don't think is unfair.
- Instead of an interval-based autosave I run a comparison on each new `prop`, which in my testing ties pretty well with actual edits.
- Adds a local storage utility and includes [store2](https://github.com/nbubna/store). 

One nice thing about this approach is that it doesn't touch `components/map` at all. @drewbo let me know what you think, this could definitely use some scrutiny.

### CODE REVIEW

File: app/scripts/components/auto-save.js [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+  renderCached: function (cached) {
+    const items = cached.map(action => (
+      <li key={action.id}><strong>{this.describeAction(action)}</strong>: {action.id}</li>
+    ));
+    return (
+      <div>
+        <ul className='cached'>{items}</ul>
+        <button onClick={this.restore}>Restore</button>
+        <button onClick={this.forget}>Forget about it</button>
```
Comment (drewbo) [inline app/scripts/components/auto-save.js]: 😂

File: app/scripts/components/auto-save.js [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+      </div>
+    );
+  },
+
+  render: function () {
+    const { cached } = this.props.save;
+    return (
+      <div className='autosave'>
+        { cached ? <div className='modal__cover'></div> : null }
```
Comment (drewbo) [inline app/scripts/components/auto-save.js]: Can these two elements be part of the same ternary check or does that throw a React single element violation?

File: app/scripts/components/auto-save.js (hunk 1 above)
Comment (kamicut) [inline app/scripts/components/auto-save.js]: 👍

File: app/scripts/components/auto-save.js (hunk 2 above)
Comment (dereklieu) [inline app/scripts/components/auto-save.js]: Yeah it's the single element violation thing. Also I've used this sort of structure to toggle the modal cover and just apply a class to the actual modal, so it animates into view.

**Pull request merged**